docs/OSD Profiles.md

```diff
@@ -0,0 +1,88 @@
+
+
+###### OPTION 2: Use the CLI to select an OSD profile (i.e. not selecting the OSD profile with your radio)
+1. Open the CLI.
+2. Type ```osdprofile``` followed by enter to display the currently selected profile.
+3. Type ```osdprofile``` followed by the profile number to be selected and press enter.  (osdprofile 0 means all configured elements will be displayed.)
+4. Type ```save``` followed by enter to save the selected profile.
+
+
+###### CONFIGURE OSD PROFILES
+The steps below are only required if using The Configurator 10.4.0 or earlier.
+1. Open the CLI.
+2. Find the name of the element to be configured, all elements can be displayed using the ```dump``` command or ```get osd``` command.  Element names end in _pos.
+3. Take the current value of the element, apply the OSD Profile config and set its new value using the ```set``` command.  The OSD Profile config value is described below.
+
+   eg. osd_vbat_pos = 6560
+       ```set osd_vbat_pos = 31136```
+       ```save```
+
+4. Repeat steps 2 & 3 for every element to be configured.
+
+
+###### OSD PROFILE CONFIG VALUES
+The OSD Profile config is stored in the higher order bits, bits 12, 13 & 14 of an OSD element, eg. osd_vbat_pos.  This means the initial value, after being configured in the OSD tab needs to be adjusted to add the OSD Profile configuration of the element.  To calculate the new value, take the current (initially configured value) of the OSD element and convert it to HEX.  Determine which OSD Profiles this element should be visible on and select the corresponding HEX value from the table below.  Perform an OR operation between the initial OSD element value (in HEX) and the HEX value from the table.  Convert the result to decimal and set the OSD element to this value using the ```set``` command in the CLI.
+
+    Binary Value    HEX Value    Profile Configuration   Description
+    14 13 12
+     0  0  0        0x0000   -   0                       OSD Profile 0, i.e. no profiles used
+     0  0  1        0x1000   -   1                       Visible in OSD Profile 1
+     0  1  0        0x2000   -   2                       Visible in OSD Profile 2
+     0  1  1        0x3000   -   1 and 2                 Visible in OSD Profile 1 and 2 only
+     1  0  0        0x4000   -   3                       Visible in OSD Profile 3 only
+     1  0  1        0x5000   -   1 and 3                 Visible in OSD Profile 1 and 3 only
+     1  1  0        0x6000   -   2 and 3                 Visible in OSD Profile 2 and 3 only
+     1  1  1        0x7000   -   1, 2 and 3              Visible in all OSD Profiles
+
+Example:
+1. osd_vbat_pos = 6560
+2. Convert 6560 to HEX results in 19A0 (HEX)  (All these operations can be done with the windows calculator)
```

Comment:
0x800 is the visible bit.

Reply:
I think we are overspecifying here: 'OSD profile 0' / not visible in any profiles is equivalent to 'not visible'.

We can save 1 bit out of the flagset (and get a more unambiguos way to encode things) if we use the existing 'visible' bit for profile 1, and define that profile 1 is the one to be used by default unless the user changes the OSD profile. This will allow for a seamless transition between one profile and multiple.

Reply:
Good idea - done.